### DISCUSSION TITLE
Why does `Span::get_hours()` return `295217`, but `Span::get_days()` return `0`?

### DISCUSSION THREAD

**Discussion by azzamsa** [BurntSushi/jiff#68] [Q&A] [answered]
Hi, 👋🏽 

I don't think 295217 hours is 0 days.

```rust
    let now = Zoned::now();
    let birthday = date(1990, 11, 28).intz("Asia/Jakarta").unwrap();
    let age = now.since(&birthday).unwrap();
    dbg!(&now);
    dbg!(&age);
    dbg!(&age.get_days());
    dbg!(&age.get_hours());
```

```rust
[jiff-demo/src/main.rs:10:5] &now = 2024-08-01T17:47:21.950643217+07:00[Asia/Jakarta]
[jiff-demo/src/main.rs:11:5] &age = PT295217h47m21.950643217s
[jiff-demo/src/main.rs:12:5] &age.get_days() = 0
[jiff-demo/src/main.rs:13:5] &age.get_hours() = 295217
```

Am I missing something?

**Comment by fbstj** [comment, marked as answer]
as the line indicates, the `age` object contains only the 295217h and zero days. you might be looking for the [round](https://docs.rs/jiff/latest/jiff/struct.Span.html#method.round) method to rebalance the object, or the [total](https://docs.rs/jiff/latest/jiff/struct.Span.html#method.total) method to calculate only the days (including fractional parts).

if you're asking why the `age` object is mostly hours, that is described in the [since](https://docs.rs/jiff/latest/jiff/struct.Zoned.html#method.since) method documentation

**Comment by BurntSushi** [comment]
@fbstj Has it right. A `Span` isn't an absolute duration like `std::time::Duration`. It's way more powerful/flexible than that. It keeps track of the individual units and is DST safe. In this case, it's preventing you from making the error that every day is 24 hours. (In `Asia/Jakarta` specifically, that is true for any recent datetime since there's no DST there and the last change to the UTC offset was in 1964. But this isn't true generally. `2024-03-10` in New York, for example, was only 23 hours long.)

I saw someone else had this same exact confusion though. So I beefed up the docs on each of `Span`'s accessors. For example: https://docs.rs/jiff/latest/jiff/struct.Span.html#method.get_hours

Is there something else that I could add to the docs that would have helped avoid this confusion?

**Reply by azzamsa** [reply]
> Is there something else that I could add to the docs that would have helped avoid this confusion?

No, all good. I think it is just a matter of familiarity. As for now, I have ported ALL my crates to Jiff without any issues on my side.
